broadinstitute/single_cell_portal_core
# Differential expression MVP (SCP-4321)

This matures the prototype differential expression (DE) UX from #1486 into a minimally viable product.

Previously, the prototype supported exactly one annotation and had a slightly clunky UX.  Now the DE feature supports multiple annotations that have manually-validated DE analyses: users can pick from 1 of 6 such annotations.  Beyond the expanded data, the UX now has lower interaction cost, greater accessibility, more intuitive and robust layout, richer analytics, more refined code, and an automated test.  These help towards [previously outlined](https://github.com/broadinstitute/single_cell_portal_core/pull/1486) scientific product goals.

Here's how it looks.

https://user-images.githubusercontent.com/1334561/169854738-bfa4c08c-9a5e-47db-bbbd-a68bfd897e13.mov

If you haven't run manual tests from #1486:
1.  Load [human genome annotation in species metadata](https://broadinstitute.slack.com/archives/CBEHTH601/p1651603466458739?thread_ts=1651603449.817109&cid=CBEHTH601), if you haven't already
2.  Start delayed job runner: execute `rails jobs:work` in a terminal
3.  Run migration: `rake db:migrate`
4.  Load public test DE study: `echo "SyntheticStudyPopulator.populate('human_milk_de_49k_cell')" | bundle exec rails console -e development`.
5.  In output of above, copy the `bucket_id` value
6.  Continue in steps below, but omit the first

If you have run manual tests from #1486:
1.  Get the bucket ID for "Human milk - differential expression" study in your local SCP
2.  Remove old prototype DE data: `rm -r gs://<your local DE study's bucket_id>/_scp_internal*` (it's OK if this errs)
3. Copy the reference public test DE study's `_scp_internal` directory to your new study's bucket: `gsutil cp -r "gs://broad-singlecellportal-public/test/studies/SCP1671/_scp_internal" "gs://<your local DE study's bucket_id>"`
4.  Sign in to your local SCP, open menu at top-right of page, select "Feature flags"
9.  Find your local DE study, set `differential_expression_frontend` feature flag to `true`
10. Sign out
9.  Go to Study Overview for your test DE study, "Human milk - differential expression"
10. Go to annotation "cell_type__ontology_label"
11.  Go to Explore tab, refresh, click "Differential expression" button
13.  In resulting modal, select group "epithelial cell"
14. Confirm that a table of genes is shown in right panel
15. Click gene "CSN2" in DE panel
16. Confirm CSN2 gene is searched
17. If steps CSN2 doesn't work, then it might be because ingest hasn't completed; try selecting group "macrophage" and searching gene "APLP2" instead.

This satisfies SCP-4321.
